README.md
---
title: "theme_agile"
output: html_document
---

This is an outline of the core functionality of theme_agile.


```{r setup, include=FALSE}
theme_agile <- function(base_size = 12, base_family = "Arial", plot.type = "formal", lines.lwd = 0.25, ticks.type = "outer", plot.grid = TRUE, axis.font = base_family, title.size = base_size*1.2, legend.size = base_size,
											 bg.col = ifelse(plot.type == "formal", "white", "#F0F0F0"), title.font = base_family , base.col = "black", axis.lines = TRUE,
											 minor.grid = ifelse(plot.grid, TRUE, FALSE), vert.grid = ifelse(plot.grid, TRUE, FALSE), ticks.length = ifelse(ticks.type == "outer", 0.2, -0.2), horz.grid = ifelse(plot.grid, TRUE, FALSE), alpha.leg = 0.1, bord.size = 0,
											 legend.bg = ifelse(plot.type == "formal", "white", "#F0F0F0"), strip.bg = ifelse(plot.type == "formal", "white", "grey80")){
	theme_bw()+
		ggplot2::theme(
		# Plot margins and finally line annotations
		plot.margin = grid::unit(c(1, 1, .5, .7), "cm"),

 		text = ggplot2::element_text(family = base_family, size = base_size),
		axis.line =  element_line(size = ifelse(axis.lines, grid::unit(lines.lwd, "mm"),0), color = "black"),
		axis.ticks.length = grid::unit(ticks.length, "cm"),
		axis.ticks.margin = grid::unit(ifelse(ticks.length > 0,0.25, -ticks.length + 0.25) , "cm"),
		axis.text.x = ggplot2::element_text(size = base_size, colour = base.col, family = axis.font),
		axis.text.y = ggplot2::element_text(size = base_size, colour = base.col, family = axis.font),
		axis.title.y = ggplot2::element_text(size =  base_size, colour = base.col, vjust = 1.5, family = axis.font),
		axis.title.x = ggplot2::element_text(size = base_size,colour = base.col,vjust = -.5, family = axis.font),
		panel.background = ggplot2::element_rect(fill = bg.col),
		plot.background = ggplot2::element_rect(fill = bg.col),
		panel.border = ggplot2::element_rect(colour = "black", fill=NA, size = bord.size),
		panel.grid.major.x = ggplot2::element_line(colour = ifelse(vert.grid, "grey60",bg.col), size = ifelse(vert.grid,0.45, 0)),
		panel.grid.minor.x = ggplot2::element_line(colour = ifelse(vert.grid, ifelse(minor.grid, "grey80",bg.col),bg.col), size = ifelse(vert.grid,0.35, 0)),
		panel.grid.major.y = ggplot2::element_line(colour = ifelse(horz.grid, "grey60",bg.col), size = ifelse(horz.grid,0.45, 0)),
		panel.grid.minor.y = ggplot2::element_line(colour = ifelse(horz.grid, ifelse(minor.grid, "grey80",bg.col),bg.col), size = ifelse(horz.grid,0.35, 0)),
		panel.grid.major = ggplot2::element_line(colour = "grey40", size=0.45),
		plot.title = ggplot2::element_text(face="bold",hjust = ifelse(plot.type == "formal", 0.5, 0) ,vjust = 2, colour = base.col, size = title.size, family = title.font),
		legend.background = ggplot2::element_rect(fill = scales::alpha(legend.bg, alpha.leg)), legend.key = ggplot2::element_blank(),
		legend.text = ggplot2::element_text(size = legend.size),
		legend.title = element_blank(),
		strip.background = ggplot2::element_rect(fill = strip.bg),
		strip.text.x = ggplot2::element_text(size = base_size + 1),
		strip.text.y = ggplot2::element_text(size = base_size + 1)
	)
}


```

## Some core theme_agile functionality

theme_agile is a ggplot2 theme designed to make journal ready figures easier, but also to make nice figures for a general audience.

I will use the mtcars data set to show the core functionality of theme_agile.

First, a plot with the defaults

## Including Plots

You can also embed plots, for example:

```{r echo=TRUE, warning= FALSE, message=FALSE}
require(ggplot2)


ggplot(mtcars, aes(mpg, disp))+
	geom_point()+
	theme_agile()

```

We may not want to have a minor grid, so let's remove that. This is done by simply setting minor.grid = FALSE.


```{r echo=TRUE, warning= FALSE, message=FALSE}

ggplot(mtcars, aes(mpg, disp))+
	geom_point()+
	theme_agile(minor.grid = FALSE)

```


If we want to get rid off the vertical grid we can simply set vert.grid = FALSE.


```{r echo=TRUE, warning= FALSE, message=FALSE}

ggplot(mtcars, aes(mpg, disp))+
	geom_point()+
	theme_agile(vert.grid = FALSE)

```


Now, let's consider the problem of creating plots that are acceptable for scientific journals.

Imagine we have a journal that does not allow grids and also requires inner axis ticks. We can do this quite easily.

```{r echo=TRUE, warning= FALSE, message=FALSE}

ggplot(mtcars, aes(mpg, disp))+
	geom_point()+
	theme_agile(plot.grid = FALSE, ticks.type = "inner")

```



```{r, include=FALSE}
   # add this chunk to end of mycode.rmd
   file.rename(from="Readme.rmd", 
               to="README.md")
```


News.md

## 18/01/2016

Some changes have been made to make the theme behave a bit better and to be consistent with v. 2 of ggplot2.

1. plot_type has been removed as an option. It was really just letting you do something that could be done easily anyway.
2. The ability to easily remove axis ticks has been added with the ticks_xy option.
3. ticks_type has been changed because axis.ticks.margin is being deprecated.
4. ticks_length is no longer something that can be modified.
5. Some minor aesthetic changes have been made to make the defaults look a bit better.


## 9/10/2015

theme_agile is now updated in response to some of the feedback after I put it on GitHub a week ago.

1. I have switched from . to _ in the theme options to be consistent. Having base_size and plot.type was probably confusing.

2. The base fone size has been switched to 11 to match the default ggplot2 size.

3. I have added in an option (grid_thick) to make grid lines thinner or thicker.


themes.R
# Theme used to create journal ready figures easilys
#
# base_size Font size
# base_family Font used for everything, unless axis fonts etc. are defined
# lines_lwd = width of lines
# plot_grid = Do you want a grid?
# font_type Font type
# title_size = font size of title
# legend_size = font size of legend
# bg_col = background colour
# title_font = font use for title
# base_col  = font colour
# horz_grid = Do you want a horizontal grid?
# bord_size = width of a rectangular border
# alpha_leg = opacity of the legend. 0 = totally transparent
# strip_bg = colour background for facets
# grid_thick = A multiplier to apply to the grid lines. 0.8 would reduce thickness by 20%
# grid_type = Grid type. Default is a solid line
# ticks_xy = Do you want ticks on the x or y axis? "x" = x-axis only, "y" = y-axis only, "xy" = both axes.
# grid_cols = Colour of the grid. 2 element vector. First element is major grid colour. If only one element, the first will be used for minor grid.



theme_agile <- function(base_size = 11, base_family = "Arial", lines_lwd = 0.50, plot_grid = TRUE, axis_font = base_family, title_size = base_size*1.2, legend_size = base_size,
												bg_col = "white",title_font = base_family , base_col  = "black", axis_lines = TRUE,
												minor_grid = ifelse(plot_grid, TRUE, FALSE), vert_grid = ifelse(plot_grid, TRUE, FALSE), ticks_type = "outer", horz_grid = ifelse(plot_grid, TRUE, FALSE), alpha_leg = 0.1, bord_size = 0,
												legend_bg = "white", strip_bg = "white", grid_thick = 1,
												grid_type = "solid", ticks_xy  = "xy", grid_cols = c("grey50", "grey70")){
	theme_bw()+
		ggplot2::theme(
			plot.margin = grid::unit(c(1, 1, .5, .7), "cm"),
			text = ggplot2::element_text(family = base_family, size = base_size),
			axis.line =  element_line(size = ifelse(axis_lines, grid::unit(lines_lwd, "mm"),0), color = "black"),
			axis.ticks.length = grid::unit(ifelse(ticks_type == "outer", 0.15, -0.15), "cm"),
			axis.ticks.x =  element_line(size = ifelse(stringr::str_detect(ticks_xy, "x"), grid::unit(lines_lwd, "cm"),0), color = "black"),
			axis.ticks.y =  element_line(size = ifelse(stringr::str_detect(ticks_xy, "y"), grid::unit(lines_lwd, "cm") ,0), color = "black"),
			axis.text.x = ggplot2::element_text(size = base_size, colour = base_col , family = axis_font,margin=margin(ifelse(ticks_type == "inner", 11, 5),5,10,5,"pt")),
			axis.text.y = ggplot2::element_text(size = base_size, colour = base_col , family = axis_font, margin=margin(5,ifelse(ticks_type == "inner", 11, 5),10,5,"pt")),
			axis.title.y = ggplot2::element_text(size =  base_size, colour = base_col , vjust = 1.5, family = axis_font),
			axis.title.x = ggplot2::element_text(size = base_size,colour = base_col ,vjust = -.5, family = axis_font),
			panel.background = ggplot2::element_rect(fill = bg_col),
			plot.background = ggplot2::element_rect(fill = bg_col),
			panel.border = ggplot2::element_rect(colour = "black", fill=NA, size = bord_size),
			panel.grid.major.x = ggplot2::element_line(linetype = grid_type,colour = ifelse(vert_grid, grid_cols[1],bg_col), size = ifelse(vert_grid,0.25 * grid_thick, 0)),
			panel.grid.minor.x = ggplot2::element_line(linetype = grid_type,colour = ifelse(vert_grid, ifelse(minor_grid, grid_cols[2 - (length(grid_cols) == 1)   ],bg_col),bg_col), size = ifelse(vert_grid,0.15* grid_thick, 0)),
			panel.grid.major.y = ggplot2::element_line(linetype = grid_type,colour = ifelse(horz_grid, grid_cols[1],bg_col), size = ifelse(horz_grid,0.25* grid_thick, 0)),
			panel.grid.minor.y = ggplot2::element_line(linetype = grid_type,colour = ifelse(horz_grid, ifelse(minor_grid, grid_cols[2 - (length(grid_cols) == 1)  ],bg_col),bg_col), size = ifelse(horz_grid,0.15* grid_thick, 0)),
			plot.title = ggplot2::element_text(face="bold",vjust = 2, colour = base_col , size = title_size, family = title_font),
			legend.background = ggplot2::element_rect(fill = scales::alpha(legend_bg, alpha_leg)), legend.key = ggplot2::element_blank(),
			legend.text = ggplot2::element_text(size = legend_size, family = base_family),
			legend.title = element_blank(),
			strip.background =  ggplot2::element_rect(colour = strip_bg, fill = strip_bg),
			strip.text.x = ggplot2::element_text(size = base_size + 1),
			strip.text.y = ggplot2::element_text(size = base_size + 1)
		)
}
